### PR TITLE
[TEVA-4375] Update Vacancy Review Page

### PR DESCRIPTION
**Pull request by josephhull676**
## Jira ticket URL

https://dfedigital.atlassian.net/browse/TEVA-4375

### Published

Screenshot 2022-09-05 at 17 13 21

### Complete Draft

Screenshot 2022-09-05 at 17 13 35

### Incomplete Draft

Screenshot 2022-09-05 at 17 13 51

### Scheduled

Screenshot 2022-09-05 at 17 14 09

### Closed

Screenshot 2022-09-05 at 17 14 25

### Preview

Screenshot 2022-09-05 at 17 05 53

### CODE REVIEW

File: app/assets/stylesheets/components/vacancy_review_heading.scss [hunk 1]
```diff
@@ -0,0 +1,14 @@
+@import 'base';
+
+.vacancy-review-heading-component {
```
Comment (alexbowen) [inline app/assets/stylesheets/components/vacancy_review_heading.scss:3]: this naming should really follow the 'root' component i.e `review-component__heading` and then you can use ampersand notation `&__heading`

File: app/assets/stylesheets/components/review.scss [hunk 2]
```diff
@@ -1,4 +1,5 @@
 @import 'base';
+@import 'vacancy_review_heading';
```
Comment (alexbowen) [inline app/assets/stylesheets/components/review.scss:2]: and put this import within review-component i.e directly after line 4

File: app/assets/stylesheets/components/vacancy_review_heading.scss (hunk 1 above)
Comment (alexbowen) [inline app/assets/stylesheets/components/vacancy_review_heading.scss:3]: this should follow the naming of the 'root' component i.e `review-component__heading` and then can use ampersand notation `&__heading` if you relocated the import mentioned above

File: app/controllers/publishers/vacancies/application_forms_controller.rb [hunk 3]
```diff
@@ -53,17 +53,24 @@ def update_vacancy
...
+      redirect_to_next_step
     else
       render "publishers/vacancies/build/applying_for_the_job_details"
     end
   end
 
+  def redirect_to_next_step
+    return redirect_to organisation_job_path(vacancy.id) if save_and_finish_later?
+
+    if vacancy.completed_steps.exclude?("documents")
```
Comment (ddippolito) [inline app/controllers/publishers/vacancies/application_forms_controller.rb:65]: As you added this here, would it be possible to remove this from `#next_invalid_step`?

https://github.com/DFE-Digital/teaching-vacancies/blob/fadd7a813255cd54b6c410a239c0a5b4e3a48404/app/controllers/publishers/vacancies/base_controller.rb#L46

Comment (ddippolito) [reply app/controllers/publishers/vacancies/application_forms_controller.rb:65]: Or actually would the  opposite be better? As in to move this check inside `#next_invalid_step` 

Thoughts?

Comment (josephhull676) [reply app/controllers/publishers/vacancies/application_forms_controller.rb]: Take a look at the latest commit I've pushed. Hopefully, this should make it all a bit clearer/nicer.

File: app/controllers/publishers/vacancies/documents_controller.rb [hunk 4]
```diff
@@ -3,7 +3,7 @@
 class Publishers::Vacancies::DocumentsController < Publishers::Vacancies::BaseController
   helper_method :form
 
-  before_action :redirect_to_next_step, only: %i[create]
+  before_action :complete_step, :unless => :document_added?, only: %i[create]
```
Comment (cesidio) [inline app/controllers/publishers/vacancies/documents_controller.rb:6]: a bit of an old notation here, can't you just do
`unless: :document_added?`

File: app/helpers/components_helper.rb [hunk 5]
```diff
@@ -16,6 +16,7 @@ module ComponentsHelper
     supportal_table: "SupportalTableComponent",
     tabs: "TabsComponent",
     vacancy_form_page_heading: "VacancyFormPageHeadingComponent",
+    vacancy_review_heading: "VacancyReviewHeadingComponent",
```
Comment (cesidio) [inline app/helpers/components_helper.rb:19]: do we still have this?

File: app/models/vacancy.rb [hunk 6]
```diff
@@ -117,10 +117,19 @@ def school_phases
...
+  def draft!
+    self.publish_on = nil
+    super
+  end
+
   def central_office?
     organisations.one? && organisations.first.trust?
   end
 
+  def live?
```
Comment (cesidio) [inline app/models/vacancy.rb:129]: you can just use `listed?` here, it does the same exact thing

File: app/helpers/components_helper.rb (hunk 5 above)
Comment (josephhull676) [reply app/helpers/components_helper.rb:19]: Ah nah, forgot to remove this 👍

File: app/controllers/publishers/vacancies/build_controller.rb [hunk 7]
```diff
@@ -23,10 +23,14 @@ def show
   def update
     if form.valid?
       update_vacancy
-      if session[:current_step] == :review
-        update_listing
+      if all_steps_valid? || save_and_finish_later?
+        if vacancy.published?
+          redirect_to organisation_job_path(vacancy.id), success: t("publishers.vacancies.show.success")
```
Comment (cesidio) [inline app/controllers/publishers/vacancies/build_controller.rb:28]: redirect_to organisation_job_path(vacancy.id), success: (t("publishers.vacancies.show.success") if vacancy.published? )

Comment (cesidio) [reply app/controllers/publishers/vacancies/build_controller.rb:28]: also check if you can put this logic into `redirect_to_next_step` method

**Review by cesidio — APPROVED** [review]

File: app/controllers/publishers/vacancies/documents_controller.rb [hunk 8]
```diff
@@ -3,15 +3,15 @@
...
+  before_action :complete_step, unless: :document_added?, only: %i[create]
 
   def create
     form.valid_documents.each do |document|
       vacancy.supporting_documents.attach(document)
       send_event(:supporting_document_created, document.original_filename, document.size, document.content_type)
     end
 
-    render :show
+    redirect_to organisation_job_build_path(vacancy.id, :documents, back_to_review: params[:publishers_job_listing_documents_form][:back_to_review])
```
Comment (cesidio) [inline app/controllers/publishers/vacancies/documents_controller.rb:14]: not sure I understand this line

Comment (cesidio) [reply app/controllers/publishers/vacancies/documents_controller.rb:14]: I guess you redirect to preserve params[:back_to_review]

Comment (josephhull676) [reply app/controllers/publishers/vacancies/documents_controller.rb]: Yeah. Have a since changed this cos it meant that errors weren't be displayed

**Pull request merged**